client/src/components/ApplicationList.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { Job, Application } from '../lib';
+import JobApplicationCard from './JobApplicationCard';
+import Pagination from './Pagination';
+
+type ApplicationListProps = {
+  jobApplication: (Application & Job)[] | undefined;
```

Comment:
It's not always good to accept `undefined` as a prop as it makes your code more complex. Why do you allow it? Why not force the caller to send an empty array?